// @bpnetguy/webpack-inject-plugin/dist/__tests__/main.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var main_1 = require("../main");
describe('injectEntry', function () {
    it('appends to only the specified entry', function () {
        expect((0, main_1.injectEntry)(undefined, 'foo', { entryName: 'bar' })).toBe('foo');
        expect((0, main_1.injectEntry)({ foo: 'bar', bar: 'baz' }, 'added', { entryName: 'bar' })).toEqual({
            foo: 'bar',
            bar: ['added', 'baz']
        });
    });
    it('supports a filter function', function () {
        expect((0, main_1.injectEntry)({ foo: 'bar', bar: 'baz', baz: 'blah' }, 'added', {
            entryName: function (e) { return e !== 'bar'; }
        })).toEqual({
            foo: ['added', 'bar'],
            bar: 'baz',
            baz: ['added', 'blah']
        });
    });
    it('throws error for unknown filter type', function () {
        expect(function () {
            (0, main_1.injectEntry)('bar', 'foo', {
                entryName: { not: 'a function ' }
            });
        }).toThrowError();
    });
    it('respects the config for ordering', function () {
        expect((0, main_1.injectEntry)(['foo', 'bar'], 'baz', { entryOrder: main_1.ENTRY_ORDER.First })).toEqual(['baz', 'foo', 'bar']);
        expect((0, main_1.injectEntry)(['foo', 'bar'], 'baz', { entryOrder: main_1.ENTRY_ORDER.Last })).toEqual(['foo', 'bar', 'baz']);
        expect((0, main_1.injectEntry)(['foo', 'bar'], 'baz', { entryOrder: main_1.ENTRY_ORDER.NotLast })).toEqual(['foo', 'baz', 'bar']);
    });
    it('order config for strings', function () {
        expect((0, main_1.injectEntry)('original', 'new', { entryOrder: main_1.ENTRY_ORDER.First })).toEqual(['new', 'original']);
        expect((0, main_1.injectEntry)('original', 'new', { entryOrder: main_1.ENTRY_ORDER.Last })).toEqual(['original', 'new']);
        expect((0, main_1.injectEntry)('original', 'new', { entryOrder: main_1.ENTRY_ORDER.NotLast })).toEqual(['new', 'original']);
    });
});
